### ISSUE TITLE
[BUG] 6.0.0 Fatal Error (Refusing to Launch)

### ISSUE THREAD

**Issue by NominalWasTaken** [radj307/volume-control#79]
### Version

6.0.0

### Windows Version

Windows 11 Home Version 22H2

### Actual Behaviour

I used then installer file and attempting to open the aplication does nothing at all no system tray icon or log file but if I launch it as administrator it will show up in my system tray and make a log file. Using the show/bring to front buttons from the system tray icon will create an icon in the taskbar but nothing will appear on screen. Hovering over the icon shows a blank window (see attached image) 
![image](https://github.com/radj307/volume-control/assets/134882520/24344e47-4d8b-4fd3-a676-9141e732a3c2)
Although the log says that a settings file will be created no file is created (see attached image)
![image](https://github.com/radj307/volume-control/assets/134882520/323011e7-aac2-46f9-8298-f6d2d2a2a1bc)

### Log

```shell
HH:mm:ss:fff           === Log Initialized @ Wednesday, September 20, 2023 11:34:23 PM ===  { Filter: 30 (ALL_EXCEPT_DEBUG) }
17:34:23:399  [INFO]   The version number in the settings file was 0.0.0, settings will be upgraded to 6.0.0.
17:34:24:156  [ERROR]  An unhandled exception occurred!
                       Sender: 'System.Windows.Threading.Dispatcher' (System.Windows.Threading.Dispatcher)
                       {
                         'Message': 'The invocation of the constructor on type 'VolumeControl.ViewModels.VolumeControlVM' that matches the specified binding constraints threw an exception.'
                         'HResult': '-2146233087'
                         'Source': 'PresentationFramework'
                         'TargetSite': {
                           'Name': 'RewrapException'
                           'DeclaringType': 'System.Windows.Markup.XamlReader'
                           'Attributes': 'PrivateScope, Assembly, Static, HideBySig'
                           'CallingConvention': 'Standard'
                         }
                         'Data':{
                           'System.Object': ''
                         }
                         'StackTrace': {
                           [0] at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
                           [1] at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
                           [2] at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
                           [3] at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
                           [4] at System.Windows.Application.LoadBamlStreamWithSyncInfo(Stream stream, ParserContext pc)
                           [5] at System.Windows.Application.LoadComponent(Uri resourceLocator, Boolean bSkipJournaledProperties)
                           [6] at System.Windows.Application.DoStartup()
                           [7] at System.Windows.Application.<.ctor>b__1_0(Object unused)
                           [8] at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
                           [9] at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
                         }
                         'InnerException': {
                           'Message': 'Process has exited, so the requested information is not available.'
                           'HResult': '-2146233079'
                           'Source': 'System.Diagnostics.Process'
                           'TargetSite': {
                             'Name': 'EnsureState'
                             'DeclaringType': 'System.Diagnostics.Process'
                             'Attributes': 'PrivateScope, Private, HideBySig'
                             'CallingConvention': 'Standard, HasThis'
                           }
                           'StackTrace': {
                             [0] at System.Diagnostics.Process.EnsureState(State state)
                             [1] at System.Diagnostics.Process.get_ProcessName()
                             [2] at VolumeControl.CoreAudio.AudioSession..ctor(AudioDevice owningDevice, AudioSessionControl2 audioSessionControl2)
                             [3] at VolumeControl.CoreAudio.AudioDeviceSessionManager.CreateAndAddSessionIfUnique(AudioSessionControl2 audioSessionControl)
                             [4] at VolumeControl.CoreAudio.AudioDeviceSessionManager..ctor(AudioDevice audioDevice)
                             [5] at VolumeControl.CoreAudio.AudioDevice..ctor(MMDevice mmDevice)
                             [6] at VolumeControl.CoreAudio.AudioDeviceManager.CreateAndAddDeviceIfUnique(MMDevice mmDevice)
                             [7] at VolumeControl.CoreAudio.AudioDeviceManager..ctor(DataFlow deviceDataFlow, MMDeviceEnumerator deviceEnumerator)
                             [8] at VolumeControl.CoreAudio.AudioDeviceManager..ctor(DataFlow deviceDataFlow)
                             [9] at VolumeControl.ViewModels.AudioDeviceManagerVM..ctor()
                             [10] at VolumeControl.ViewModels.VolumeControlVM..ctor()
                             [11] at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
                           }
                         }
                       }
```

**Comment by radj307** [comment]
Thanks for reporting this! It gives us a much better idea of what's going on with the Windows 11 version.

This may actually be 2 bugs:
- Program Files isn't a great directory to store settings in when users are running it
- Volume Control window is blank on some Windows 11 installations
  #71 

To do:
- Move config file out of the Program Files directory when installing for all users
- Figure out what's causing the v6 GUI to not work properly on Windows 11

**Comment by radj307** [comment]
Fixed bug 1 in 1bf6d873ea77d84e96aafc76deae2127aaf14bea & 1647d3493ae0fac1f8e1bb9b858ca5523b1f7982
> Program Files isn't a great directory to store settings in when running as user instead of as administrator

- Added a new configuration `Release-ForInstaller` with a preprocessor switch that changes the default path for the config & log files to the AppData directory *(`.../AppData/Local/radj307/VolumeControl/`)*.  
- Added a new publish profile `PublishForInstaller.pubxml` that builds the binary in `Release-ForInstaller` configuration and saves it into a subdirectory. Changed the vcsetup.iss config to get the executable file from that subdirectory.
- Finally, changed the CI script for generating new releases to respect this new build workflow.

All of this is to say, when Volume Control is installed via `VolumeControl-Installer.exe`, the config & log files are stored in the user's AppData directory rather than wherever the executable was installed to. The portable version continues to save the config & log files in the same directory as the executable.

**Comment by radj307** [comment]
As for the next bug, I'll test the application on a Windows 11 Home 22H2 VM and do some more investigating tonight or tomorrow.

**Comment by radj307** [comment]
Since this problem was mentioned in another issue report (#82), I'm going to push forward with releasing the changes I have so far as v6.0.1 since I cannot replicate the second bug.

**Comment by radj307** [comment]
[v6.0.1](https://github.com/radj307/volume-control/releases/tag/6.0.1) is now available

**Comment by radj307** [comment]
Closing as main problem is fixed.

**Comment by radj307** [comment]
This issue may have actually been present in v6.0.1 - v6.0.3 due to another issue with a CI script.  
It has been fixed in [v6.0.4](https://github.com/radj307/volume-control/releases/tag/6.0.4)
